### PR TITLE
Add HSQLDB support and move phoenix support into an adapter

### PR DESCRIPTION
**Pull request by F21**

### CODE REVIEW

**Comment by risdenk** [comment]
For reference I tried this against the new .travis.yml in my fork and got a build error:

https://travis-ci.org/risdenk/calcite-avatica-go/builds/366966818

```
Solving failure: No versions of github.com/apache/calcite-avatica-go met constraints:
	master: Could not introduce github.com/apache/calcite-avatica-go@master due to multiple problematic subpackages:
	Subpackage github.com/apache/calcite-avatica-go/errors is missing. (Package is required by (root).)	Subpackage github.com/apache/calcite-avatica-go/generic is missing. (Package is required by (root).)	Subpackage github.com/apache/calcite-avatica-go/hsqldb is missing. (Package is required by (root).)	Subpackage github.com/apache/calcite-avatica-go/internal is missing. (Package is required by (root).)	Subpackage github.com/apache/calcite-avatica-go/phoenix is missing. (Package is required by (root).)
```

**Comment by risdenk** [comment]
Looks like this might have been a false alarm.
```
mkdir -p $HOME/gopath/src/github.com/risdenk/calcite-avatica-go
```

Travis was picking up my username for the gopath and so the packages wouldn't match. Looking into this a bit more.

One idea might be to force the go import path so that forks work. https://docs.travis-ci.com/user/languages/go/#Go-Import-Path

File: phoenix/phoenix.go [hunk 1]
```diff
@@ -1,82 +1,105 @@
...
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package avatica
+package phoenix
```
Comment (risdenk) [inline phoenix/phoenix.go]: Missing license header?

Comment (F21) [reply phoenix/phoenix.go]: Good catch! thanks!

**Pull request merged**